stub-runner/stub-runner/src/main/groovy/com/ofg/stub/registry/StubRegistry.groovy

```diff
@@ -54,7 +55,18 @@ class StubRegistry {
                 .uriSpec(URI_SPEC)
                 .address('localhost')
                 .port(stubServer.port)
-                .name(stubServer.projectMetadata.projectRelativePath)
+                .name(serviceName(stubServer))
                 .build()
     }
+
+    private static String serviceName(StubServer stubServer) {
+        ProjectMetadata metadata = stubServer.projectMetadata
+        String projectRelativePath = metadata.projectRelativePath
+        if (projectRelativePath.endsWith(metadata.projectName)) {
+            return projectRelativePath
+        } else {
+            int idx = projectRelativePath.lastIndexOf('/')
+            return projectRelativePath.substring(0, idx + 1) + metadata.projectName
```

Comment:
There could be more `/` in `projectRelativePath` so the last one need to be find and I doubt this could be simplified somehow. I extracted this to a method taking path without project name at the end.